api/app/pagination.py

```diff
@@ -1,7 +1,62 @@
+import base64
+import json
+from collections import OrderedDict
+
 from rest_framework.pagination import PageNumberPagination
+from rest_framework.response import Response
+from rest_framework.utils.urls import replace_query_param
 
 
 class CustomPagination(PageNumberPagination):
     page_size = 999
     page_size_query_param = "page_size"
     max_page_size = 999
+
+
+class IdentityPagination(CustomPagination):
+    last_evaluated_key = None
+    previous_last_evaluated_key = None
+    dynamodb_count = None
+
+    def set_pagination_state_dynamo(self, dynamo_queryset, request) -> None:
```

Comment:
I still don't understand why we can't utilise more of the inbuilt functionality of DRF here for the pagination class. Is there a way we could just extend, e.g. `rest_framework.pagination.CursorPagination` or `rest_framework.pagination.LimitOffsetPagination` instead? Seems like that would be closer to our needs and we could just implement the dynamo specific bits. 

Maybe we need to create a totally separate endpoint for this?

Reply:
The problem with this is that the `drf/django` pagination is built around the idea of `queryset` and we kind of don't have that.